### ISSUE TITLE
requirement.txt

### ISSUE THREAD

**Issue by zyt0929** [CalcuLuUus/RASM#3]
Can you give the version of the package,  for example, matplotlib = 2.3.4,scikit-image = 1.2.3.  Like other open source code.
so I can run the code in Docker

**Comment by lime-j** [comment]
Hello! Thanks for your attention to our work! Our codebase works well on various environment settings, an example of mine is listed as follows:

absl-py                  2.1.0
addict                   2.4.0
aiofiles                 23.2.1
aiohappyeyeballs         2.4.0
aiohttp                  3.10.6
aiosignal                1.3.1
annotated-types          0.7.0
antlr4-python3-runtime   4.9.3
anyio                    4.6.2.post1
attrs                    24.2.0
beautifulsoup4           4.12.3
certifi                  2024.6.2
charset-normalizer       3.3.2
click                    8.1.7
contourpy                1.2.1
cycler                   0.12.1
dataclasses              0.6
docker-pycreds           0.4.0
einops                   0.8.0
fastapi                  0.115.4
ffmpy                    0.4.0
filelock                 3.14.0
flash-attn               2.6.1
focal-frequency-loss     0.3.0
fonttools                4.53.0
frozenlist               1.4.1
fsspec                   2024.6.0
future                   1.0.0
fvcore                   0.1.5.post20221221
gdown                    5.2.0
gitdb                    4.0.11
GitPython                3.1.43
gradio                   5.5.0
gradio_client            1.4.2
grpcio                   1.64.1
h11                      0.14.0
h5py                     3.11.0
httpcore                 1.0.6
httpx                    0.27.2
huggingface-hub          0.26.2
idna                     3.7
imageio                  2.34.1
importlib_metadata       8.0.0
iopath                   0.1.10
Jinja2                   3.1.4
joblib                   1.4.2
kiwisolver               1.4.5
kornia                   0.7.3
kornia_rs                0.1.4
lazy_loader              0.4
lightning-utilities      0.11.7
linformer                0.2.3
llvmlite                 0.43.0
lmdb                     1.5.1
lpips                    0.1.4
Markdown                 3.6
markdown-it-py           3.0.0
MarkupSafe               2.1.5
matplotlib               3.9.0
mdurl                    0.1.2
mpmath                   1.3.0
multidict                6.1.0
natsort                  8.4.0
natten                   0.17.1+torch230cu121
networkx                 3.3
numba                    0.60.0
numpy                    1.26.4
nvidia-cublas-cu12       12.1.3.1
nvidia-cuda-cupti-cu12   12.1.105
nvidia-cuda-nvrtc-cu12   12.1.105
nvidia-cuda-runtime-cu12 12.1.105
nvidia-cudnn-cu12        8.9.2.26
nvidia-cufft-cu12        11.0.2.54
nvidia-curand-cu12       10.3.2.106
nvidia-cusolver-cu12     11.4.5.107
nvidia-cusparse-cu12     12.1.0.106
nvidia-nccl-cu12         2.20.5
nvidia-nvjitlink-cu12    12.5.40
nvidia-nvtx-cu12         12.1.105
omegaconf                2.3.0
opencv-python            4.10.0.82
orjson                   3.10.11
packaging                24.1
pandas                   2.2.3
pillow                   10.3.0
pip                      24.0
platformdirs             4.2.2
portalocker              2.8.2
protobuf                 4.25.3
psutil                   6.1.0
ptflops                  0.7.3
pycocotools              2.0.8
pydantic                 2.9.2
pydantic_core            2.23.4
pydub                    0.25.1
Pygments                 2.18.0
PyMatting                1.1.12
pyparsing                3.1.2
PySocks                  1.7.1
python-dateutil          2.9.0.post0
python-multipart         0.0.12
pytorch-lightning        2.4.0
pytorch-msssim           1.0.0
pytz                     2024.2
PyWavelets               1.7.0
PyYAML                   6.0.1
requests                 2.32.3
rich                     13.9.4
ruff                     0.7.3
safehttpx                0.1.1
safetensors              0.4.3
scikit-image             0.18.1
scikit-learn             1.5.0
scipy                    1.13.1
semantic-version         2.10.0
sentry-sdk               2.17.0
setproctitle             1.3.3
setuptools               69.5.1
shellingham              1.5.4
six                      1.16.0
smmap                    5.0.1
sniffio                  1.3.1
soupsieve                2.5
starlette                0.41.2
swattention              1.0
sympy                    1.12.1
tabulate                 0.9.0
tensorboard              2.17.0
tensorboard-data-server  0.7.2
termcolor                2.4.0
thop                     0.1.1.post2209072238
threadpoolctl            3.5.0
tifffile                 2024.5.22
timm                     1.0.3
tomli                    2.0.1
tomlkit                  0.12.0
torch                    2.3.1
torchvision              0.18.1
tqdm                     4.66.4
triton                   2.3.1
typer                    0.13.0
typing_extensions        4.12.2
tzdata                   2024.2
urllib3                  2.2.1
uvicorn                  0.32.0
wandb                    0.18.5
websockets               12.0
Werkzeug                 3.0.3
wheel                    0.43.0
yacs                     0.1.8
yapf                     0.40.2
yarl                     1.12.1
zipp                     3.19.2

**Comment by CalcuLuUus** [comment]
> Where is the structural diagram of the RA module? I couldn't find it in the paper.

Hello! The RA module, i.e., Regional Attention module, can be seen in Fig 1(c) of the paper, where it is schematically compared with Self Attention and Window Attention.

**Comment by zyt0929** [comment]
I saw it, it's just a simple diagram. What I'm talking about is the framework diagram, which means others can reproduce the code based on your framework diagram. This diagram was not provided in your paper.

**Comment by zyt0929** [comment]
I cannot know how the code of the RA module is completed based solely on Fig 1 (c)

**Comment by CalcuLuUus** [comment]
> I saw it, it's just a simple diagram. What I'm talking about is the framework diagram, which means others can reproduce the code based on your framework diagram. This diagram was not provided in your paper.

We were inspired by Neighborhood Attention Transformer when implementing Regional Attention, you can see  [Neighborhood Attention Transformer](https://github.com/SHI-Labs/Neighborhood-Attention-Transformer) & [NATTEN](https://github.com/SHI-Labs/NATTEN) for more details.

**Comment by CalcuLuUus** [comment]
> Hello! Thanks for your attention to our work! Our codebase works well on various environment settings, an example of mine is listed as follows:
> 
> absl-py 2.1.0 addict 2.4.0 aiofiles 23.2.1 aiohappyeyeballs 2.4.0 aiohttp 3.10.6 aiosignal 1.3.1 annotated-types 0.7.0 antlr4-python3-runtime 4.9.3 anyio 4.6.2.post1 attrs 24.2.0 beautifulsoup4 4.12.3 certifi 2024.6.2 charset-normalizer 3.3.2 click 8.1.7 contourpy 1.2.1 cycler 0.12.1 dataclasses 0.6 docker-pycreds 0.4.0 einops 0.8.0 fastapi 0.115.4 ffmpy 0.4.0 filelock 3.14.0 flash-attn 2.6.1 focal-frequency-loss 0.3.0 fonttools 4.53.0 frozenlist 1.4.1 fsspec 2024.6.0 future 1.0.0 fvcore 0.1.5.post20221221 gdown 5.2.0 gitdb 4.0.11 GitPython 3.1.43 gradio 5.5.0 gradio_client 1.4.2 grpcio 1.64.1 h11 0.14.0 h5py 3.11.0 httpcore 1.0.6 httpx 0.27.2 huggingface-hub 0.26.2 idna 3.7 imageio 2.34.1 importlib_metadata 8.0.0 iopath 0.1.10 Jinja2 3.1.4 joblib 1.4.2 kiwisolver 1.4.5 kornia 0.7.3 kornia_rs 0.1.4 lazy_loader 0.4 lightning-utilities 0.11.7 linformer 0.2.3 llvmlite 0.43.0 lmdb 1.5.1 lpips 0.1.4 Markdown 3.6 markdown-it-py 3.0.0 MarkupSafe 2.1.5 matplotlib 3.9.0 mdurl 0.1.2 mpmath 1.3.0 multidict 6.1.0 natsort 8.4.0 natten 0.17.1+torch230cu121 networkx 3.3 numba 0.60.0 numpy 1.26.4 nvidia-cublas-cu12 12.1.3.1 nvidia-cuda-cupti-cu12 12.1.105 nvidia-cuda-nvrtc-cu12 12.1.105 nvidia-cuda-runtime-cu12 12.1.105 nvidia-cudnn-cu12 8.9.2.26 nvidia-cufft-cu12 11.0.2.54 nvidia-curand-cu12 10.3.2.106 nvidia-cusolver-cu12 11.4.5.107 nvidia-cusparse-cu12 12.1.0.106 nvidia-nccl-cu12 2.20.5 nvidia-nvjitlink-cu12 12.5.40 nvidia-nvtx-cu12 12.1.105 omegaconf 2.3.0 opencv-python 4.10.0.82 orjson 3.10.11 packaging 24.1 pandas 2.2.3 pillow 10.3.0 pip 24.0 platformdirs 4.2.2 portalocker 2.8.2 protobuf 4.25.3 psutil 6.1.0 ptflops 0.7.3 pycocotools 2.0.8 pydantic 2.9.2 pydantic_core 2.23.4 pydub 0.25.1 Pygments 2.18.0 PyMatting 1.1.12 pyparsing 3.1.2 PySocks 1.7.1 python-dateutil 2.9.0.post0 python-multipart 0.0.12 pytorch-lightning 2.4.0 pytorch-msssim 1.0.0 pytz 2024.2 PyWavelets 1.7.0 PyYAML 6.0.1 requests 2.32.3 rich 13.9.4 ruff 0.7.3 safehttpx 0.1.1 safetensors 0.4.3 scikit-image 0.18.1 scikit-learn 1.5.0 scipy 1.13.1 semantic-version 2.10.0 sentry-sdk 2.17.0 setproctitle 1.3.3 setuptools 69.5.1 shellingham 1.5.4 six 1.16.0 smmap 5.0.1 sniffio 1.3.1 soupsieve 2.5 starlette 0.41.2 swattention 1.0 sympy 1.12.1 tabulate 0.9.0 tensorboard 2.17.0 tensorboard-data-server 0.7.2 termcolor 2.4.0 thop 0.1.1.post2209072238 threadpoolctl 3.5.0 tifffile 2024.5.22 timm 1.0.3 tomli 2.0.1 tomlkit 0.12.0 torch 2.3.1 torchvision 0.18.1 tqdm 4.66.4 triton 2.3.1 typer 0.13.0 typing_extensions 4.12.2 tzdata 2024.2 urllib3 2.2.1 uvicorn 0.32.0 wandb 0.18.5 websockets 12.0 Werkzeug 3.0.3 wheel 0.43.0 yacs 0.1.8 yapf 0.40.2 yarl 1.12.1 zipp 3.19.2

I am very sorry for providing the wrong information. Please note that the version of natten should be 0.15.1.
